external/kafka/src/main/scala/org/apache/spark/streaming/kafka/DirectKafkaInputDStream.scala

```diff
@@ -82,8 +83,12 @@ class DirectKafkaInputDStream[
     }
   }
 
+  private val offsetMap = new mutable.HashMap[Time, Map[TopicAndPartition, Long]]()
```

Comment:
Do you mean that we can use `batchForTime` ? Looks like this will only be updated when checkpoint is enabled.